docs: create docs for the github copilot apify plugin (#2699)
apify/apify-docs: sources/platform/integrations/ai/github-copilot.md

---
title: GitHub Copilot integration
sidebar_label: GitHub Copilot
description: Learn how to install the Apify plugin for GitHub Copilot in VS Code to discover, run, and build Actors with the Apify MCP server, skills, and a routing agent.
slug: /integrations/github-copilot
---

import ThirdPartyDisclaimer from '@site/sources/_partials/_third-party-integration.mdx';

[GitHub Copilot](https://github.com/features/copilot) is GitHub's AI coding assistant. In VS Code, its agent mode reads and edits your workspace, runs commands, and completes multi-step development tasks.

The [Apify plugin for GitHub Copilot](https://github.com/apify/apify-github-copilot-plugin) connects Copilot to Apify's library of [Actors](https://apify.com/store) and bundles:

> - The [Apify MCP server](/platform/integrations/mcp) for searching the Store, running Actors, and retrieving datasets through the [Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro).

Reviewer: Apify Store takes no article: `the Store` → `Apify Store` (matches usage on line 89).

```suggestion
- The [Apify MCP server](/platform/integrations/mcp) for searching Apify Store, running Actors, and retrieving datasets through the [Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro).
```

    ![VS Code Command Palette with the Chat: Install Plugin from Source command selected](images/github-copilot/install-plugin-command.webp)

1. Paste the repository URL into the input field and press Enter.

    ![VS Code input field with the Apify plugin repository URL pasted in](images/github-copilot/install-plugin-url.webp)

1. In the trust dialog, review the source and select **Trust**.

    ![VS Code dialog asking whether to trust plugins from the Apify repository](images/github-copilot/install-plugin-trust.webp)

1. VS Code installs the plugin and opens the **Plugin: apify** panel, which shows **Disable** and **Uninstall** actions. You can reopen it anytime from the Command Palette (`Ctrl+Shift+P`, or `Cmd+Shift+P` on macOS) with **Chat: Plugins**, which lists your installed agent plugins.

    ![VS Code Plugin: apify panel open over the Agent Plugins list, showing the installed Apify plugin](images/github-copilot/plugin-installed.webp)

1. Open Copilot Chat, open the mode picker, and select the **apify** agent.

    ![Copilot Chat mode picker with the apify agent selected](images/github-copilot/agent-picker.webp)

## Connect the Apify MCP server

> The plugin bundles the Apify MCP server (`https://mcp.apify.com/`) - its configuration ships in the plugin's `.mcp.json`, so you don't add it manually. Read-only tools like searching the Store and fetching Actor details work without signing in, but you need to authenticate to run Actors and access your account data.

Reviewer: Apify Store takes no article: `the Store` → `Apify Store`.

```suggestion
The plugin bundles the Apify MCP server (`https://mcp.apify.com/`) - its configuration ships in the plugin's `.mcp.json`, so you don't add it manually. Read-only tools like searching Apify Store and fetching Actor details work without signing in, but you need to authenticate to run Actors and access your account data.
```

> 1. Confirm the **Apify MCP Server** tool is enabled for the `apify` agent. Open the tools picker in Copilot Chat and check that **Apify MCP Server** is selected.

Reviewer: The two sentences say the same thing twice - one is enough.

```suggestion
1. Open the tools picker in Copilot Chat and confirm that **Apify MCP Server** is selected for the `apify` agent.
```

    ![Bundled .mcp.json with the Start action above the apify-mcp-server entry](images/github-copilot/mcp-start.webp)

1. VS Code prompts that the MCP server wants to authenticate to `console-backend.apify.com`. Select **Allow**, complete the Apify OAuth flow in your browser, and choose the account to connect.

    ![VS Code dialog asking to allow the apify-mcp-server to authenticate to console-backend.apify.com](images/github-copilot/mcp-oauth-allow.webp)

1. The `apify-mcp-server` entry shows **Running** and exposes its tools.

    ![Bundled .mcp.json showing the apify-mcp-server as Running with Stop and Restart actions](images/github-copilot/mcp-running.webp)

:::tip Session persistence

The connection stays authenticated for future sessions. You can revoke access at any time in [Apify Console > Settings > Integrations](https://console.apify.com/settings/integrations).

:::

## Run your first prompt

Select the **apify** agent and describe what you want in natural language. The agent routes the request to the right tool or skill, so you don't need to name tools yourself.

> > "Use Apify to find a good Actor for scraping Google Maps places. Show me the best option, its input requirements, pricing model, and what kind of dataset output it returns. Do not run the Actor yet."

Reviewer: No need to use blockquote and quotation marks.

```suggestion
> Use Apify to find a good Actor for scraping Google Maps places. Show me the best option, its input requirements, pricing model, and what kind of dataset output it returns. Do not run the Actor yet.
```

The agent searches Apify Store, fetches the top Actor's details through the Apify MCP server, and summarizes its inputs, pricing, and output - all without running the Actor.

To check what's available, ask the agent to list its Apify tools.

![Copilot Chat listing the available Apify MCP tools](images/github-copilot/verify-tools.webp)

## Bundled skills

| Skill | Description |
| --- | --- |
| `apify-ultimate-scraper` | CLI-driven extraction using existing Actors for multi-step scraping and lead-generation workflows. |
| `apify-actor-development` | Full Actor lifecycle - template selection, development, local testing, and deployment with `apify push`. |
| `apify-actorization` | Converts existing JavaScript, TypeScript, Python, or CLI projects into Apify Actors. |
| `apify-generate-output-schema` | Generates dataset and key-value store schemas for existing Actors. |
| `apify-sdk-integration` | Integrates Actor execution into applications using the `apify-client` package. |

Example prompts that route to specific skills:

_Ultimate scraper:_

> > "Find 10 highly rated coffee shops in Seattle with name, address, rating, phone, and website."

Reviewer: Same for the other example prompts below.

- **SDK integration (`apify-client`)**: reads the `APIFY_TOKEN` environment variable from your application's environment.

## Troubleshooting

### The `apify` agent doesn't appear in the picker

Confirm that the plugin is installed and trusted (rerun **Chat: Install Plugin from Source** if needed), that **Chat: Plugins** is enabled in Settings, and that you reloaded the window after installing. Plugin support requires VS Code 1.120 or newer.

### The Apify MCP server won't start

Open the plugin's `.mcp.json` and select the **Start** action above the `apify-mcp-server` entry, then complete the **Allow** prompt and account selection. Check the **Output** panel (**MCP: apify-mcp-server**) - a successful start ends with a line about the discovered tools. Confirm the **Apify MCP Server** tool is enabled for the `apify` agent in the tools picker.

### Browser doesn't open, or OAuth fails

If the browser doesn't open automatically, copy the OAuth URL from the VS Code dialog and paste it into your browser manually.

> If you're running VS Code over SSH, in a devcontainer, or in any environment without a browser, the MCP OAuth flow can't complete. Authenticate locally first so the connection is stored, or use the CLI and SDK paths instead - run `apify login`, or set `APIFY_TOKEN`:

Reviewer: VS Code's official spelling is _dev container_.

```suggestion
If you're running VS Code over SSH, in a dev container, or in any environment without a browser, the MCP OAuth flow can't complete. Authenticate locally first so the connection is stored, or use the CLI and SDK paths instead - run `apify login`, or set `APIFY_TOKEN`:
```

```bash
export APIFY_TOKEN=<YOUR_API_TOKEN>
```

> ### The wrong skill or transport keeps getting picked

Reviewer: This heading is passive/gerund - suggest an active rephrase, matching the other troubleshooting headings.

```suggestion
### The agent picks the wrong skill or transport
```
